python/lsst/pipe/tasks/skyCorrection.py

```diff
@@ -183,64 +106,100 @@ class SkyCorrectionConnections(pipeBase.PipelineTaskConnections, dimensions=("in
-        storageClass="ImageF",
-        dimensions=["instrument", "visit"],
+    camera = cT.PrerequisiteInput(
+        doc="Input camera.",
+        name="camera",
+        storageClass="Camera",
+        dimensions=["instrument"],
+        isCalibration=True,
     )
     skyCorr = cT.Output(
-        doc="Output sky corrected images.",
-        name='skyCorr',
+        doc="Sky correction data, to be subtracted from the calibrated exposures.",
+        name="skyCorr",
         multiple=True,
         storageClass="Background",
         dimensions=["instrument", "visit", "detector"],
     )
+    calExpMosaic = cT.Output(
+        doc="Full focal plane mosaicked image of the sky corrected calibrated exposures.",
+        name="calexp_skyCorr_visit_mosaic",
+        storageClass="ImageF",
+        dimensions=["instrument", "visit"],
+    )
+    calBkgMosaic = cT.Output(
+        doc="Full focal plane mosaicked image of the sky corrected calibrated exposure backgrounds.",
+        name="calexpBackground_skyCorr_visit_mosaic",
+        storageClass="ImageF",
+        dimensions=["instrument", "visit"],
+    )
 
 
-class SkyCorrectionConfig(pipeBase.PipelineTaskConfig, pipelineConnections=SkyCorrectionConnections):
-    """Configuration for SkyCorrectionTask"""
-    bgModel = ConfigField(dtype=FocalPlaneBackgroundConfig, doc="Background model")
-    bgModel2 = ConfigField(dtype=FocalPlaneBackgroundConfig, doc="2nd Background model")
-    sky = ConfigurableField(target=SkyMeasurementTask, doc="Sky measurement")
-    maskObjects = ConfigurableField(target=MaskObjectsTask, doc="Mask Objects")
-    doMaskObjects = Field(dtype=bool, default=True, doc="Mask objects to find good sky?")
-    doBgModel = Field(dtype=bool, default=True, doc="Do background model subtraction?")
-    doBgModel2 = Field(dtype=bool, default=True, doc="Do cleanup background model subtraction?")
-    doSky = Field(dtype=bool, default=True, doc="Do sky frame subtraction?")
-    binning = Field(dtype=int, default=8, doc="Binning factor for constructing focal-plane images")
-    calexpType = Field(dtype=str, default="calexp",
-                       doc="Should be set to fakes_calexp if you want to process calexps with fakes in.")
+class SkyCorrectionConfig(PipelineTaskConfig, pipelineConnections=SkyCorrectionConnections):
+    maskObjects = ConfigurableField(
+        target=MaskObjectsTask,
+        doc="Mask Objects",
+    )
+    doMaskObjects = Field(
+        dtype=bool,
+        default=True,
+        doc="Iteratively mask objects to find good sky?",
+    )
+    bgModel = ConfigField(
+        dtype=FocalPlaneBackgroundConfig,
+        doc="Initial background model, prior to sky frame subtraction",
+    )
+    doBgModel = Field(
+        dtype=bool,
+        default=True,
+        doc="Do initial background model subtraction (prior to sky frame subtraction)?",
+        deprecated="This field is deprecated and will be removed after v26. See RFC-898 for further details.",
```

Comment:
removed after v26, since v25 candidate is already tagged. Also, if this is just part of the renaming, suggest the field that takes its place.

Reply:
Here's a snippet from [the Dev Guide](https://developer.lsst.io/stack/deprecating-interfaces.html#code-removal):

> Removal of the deprecated code occurs, at the earliest, immediately after the next major release following the release with the first deprecation release note; at the latest, immediately before the major release following. In other words, if deprecation is first noted in release 17.2.3, the code cannot be removed until after 18.0 is released and must be removed before 19.0 is released.

To my mind, v25 hasn't been fully released yet (it's still only a release candidate), but assuming that still counts, that means that the next major release is v26. This means that this deprecation effort should take place in the v26 to v27 time period, no?

Reply:
Right, after v26 is correct then. Apologies.

Reply:
Also, it looks like both `doBgModel` and `doBgModel1` are deprecated. That's surely not what you intended.

Reply:
Following [RFC-898](https://jira.lsstcorp.org/browse/RFC-898), the first full focal plane background model will be made mandatory. That means that `doBgModel` will be removed on the implementation ticket. However, as we're also changing the name of the task that it targets (`bgModel`) that means that I also need to temporarily set up a new `bgModel1` config field for use from now until the deprecation is implemented. At the time of implementation, I'll remove both `doBgModel` and `doBgModel1` (and make the relevant code block mandatory).

I can't think of a better way to do this, but if you can, happy to switch it up!

Reply:
Either way, I can't see a way to not raise `DeprecatedWarning` messages. So this is fine.